### PR TITLE
Optimize JVMTI watched fields

### PR DESCRIPTION
**Pull request by gacholio**
Correct the math used to determine the index into the watched field bit
array.

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

File: runtime/oti/jvmtiInternal.h [hunk 1]
```diff
@@ -91,15 +91,15 @@ typedef struct J9JVMTIObjectTag {
...
-#define J9JVMTI_WATCHED_FIELD_ACCESS_BITS_PER_FIELD ((UDATA)2)
+#define J9JVMTI_WATCHED_FIELD_BITS_PER_FIELD 2
 #define J9JVMTI_WATCHED_FIELDS_PER_UDATA \
-	((sizeof(UDATA) * 8) / J9JVMTI_WATCHED_FIELD_ACCESS_BITS_PER_FIELD)
+	((sizeof(UDATA) * 8) / J9JVMTI_WATCHED_FIELD_BITS_PER_FIELD)
 #define J9JVMTI_CLASS_REQUIRES_ALLOCATED_J9JVMTI_WATCHED_FIELD_ACCESS_BITS(clazz) \
 	((clazz)->romClass->romFieldCount > J9JVMTI_WATCHED_FIELDS_PER_UDATA)
 #define J9JVMTI_WATCHED_FIELD_ARRAY_INDEX(index) \
-	(((UDATA)(index)) / J9JVMTI_WATCHED_FIELDS_PER_UDATA)
+	(((index)) / J9JVMTI_WATCHED_FIELDS_PER_UDATA)
```
Comment (pshipton) [inline runtime/oti/jvmtiInternal.h]: some extra brackets around (index) here, which could be removed, although they aren't harmful.

File: runtime/oti/jvmtiInternal.h [hunk 2]
```diff
@@ -91,15 +91,15 @@ typedef struct J9JVMTIObjectTag {
...
 #define J9JVMTI_WATCHED_FIELD_ACCESS_BIT(index) \
-	(((UDATA)1) << ((((UDATA)(index)) * J9JVMTI_WATCHED_FIELD_ACCESS_BITS_PER_FIELD) % J9JVMTI_WATCHED_FIELDS_PER_UDATA))
+	((UDATA)1 << (((index) % J9JVMTI_WATCHED_FIELDS_PER_UDATA) * J9JVMTI_WATCHED_FIELD_BITS_PER_FIELD))
```
Comment (keithc-ca) [inline runtime/oti/jvmtiInternal.h]: I would have preferred that you left the clarifying parentheses so no one has to guess/remember/check whether `(UDATA)1 << s` means `((UDATA)1) << s` or `(UDATA)(1 << s)`, but I won't press the issue.

**Comment by pshipton** [comment]
jenkins test sanity win

**Comment by pshipton** [comment]
JDK8 failed because of #2129. I'll go ahead and merge as the JDK10 testing passed.

**Pull request merged**